tinygrad/codegen/llvmir.py

```diff
@@ -41,10 +41,6 @@ def uops_to_llvm_ir(uops:List[UOp], bufs:List[LazyBuffer]) -> str:
   func_dtypes = [{dtypes.float16:ir.HalfType(), dtypes.float32:ir.FloatType(), dtypes.int8:ir.IntType(8), dtypes.uint8:ir.IntType(8), dtypes.bool: ir.IntType(1), dtypes.int64: ir.IntType(64), dtypes.int32: ir.IntType(32)}[buf.dtype] for buf in bufs]
   func = ir.Function(module, ir.FunctionType(ir.VoidType(), [x.as_pointer() for x in func_dtypes]), name='exec')
 
-  # force llvmlite to allow us to add function attribute then add the attribute
-  func.attributes._known = func.attributes._known.union(frozenset(['"no-nans-fp-math"="true"']))
```

Comment:
We needed this to get performance.

Reply:
Overall I totally agree this PR shouldn't be merged if performance is preferred over edge-case correctness over all backends.  This PR is just meant to show what would be required to make the backends match the CMPEQ/MAX op behavior of torch CPU: disabling fast math for CMPEQ, using select and isnan for for MAX instead of fmax.

I suppose I'm making the case that if performance is key, tinygrad should adopt the most GPU-efficient behavior as the base case and make the CPU backends less efficient to match their behavior since overall FLOPS will come from GPU.